apps/web/src/services/project/components/ProjectGroupMemberManagementModal.vue

```diff
@@ -0,0 +1,266 @@
+import { i18n } from '@/translations';
+
+import { useAllReferenceStore } from '@/store/reference/all-reference-store';
+import type { UserReferenceMap } from '@/store/reference/user-reference-store';
+
+import { showSuccessMessage } from '@/lib/helper/notice-alert-helper';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+import { useProxyValue } from '@/common/composables/proxy-state';
+
+import { useRoleFormatter } from '@/services/iam/composables/refined-table-data';
+
+
+interface SelectedUserItem extends SelectDropdownMenuItem {
+    roleImage?: string;
+    roleName?: TranslateResult;
+}
+interface Props {
+    visible?: boolean;
+    projectGroupId?: string;
+}
+const props = withDefaults(defineProps<Props>(), {
+    visible: false,
+    projectGroupId: undefined,
+});
+const emit = defineEmits<{(e: 'confirm'): void;
+}>();
+
+const allReferenceStore = useAllReferenceStore();
+const storeState = reactive({
+    users: computed<UserReferenceMap>(() => allReferenceStore.getters.user),
+});
+const state = reactive({
+    loading: false,
+    proxyVisible: useProxyValue('visible', props, emit),
+    workSpaceUserList: [] as WorkspaceUserModel[],
+    projectGroupUserIdList: [] as string[],
+    //
+    userMenuItems: computed<SelectDropdownMenuItem[]>(() => {
+        const _items: SelectDropdownMenuItem[] = [];
+        state.workSpaceUserList.forEach((user) => {
+            const singleItem = {
+                name: user.user_id,
+                label: user.name ? `${user.user_id} (${user.name})` : user.user_id,
+                disabled: false,
+            };
+            if ((state.selectedUserItems.find((d) => d.name === user.user_id))
+                || (state.projectGroupUserIdList.includes(user.user_id))) {
+                singleItem.disabled = true;
+            }
+            _items.push(singleItem);
+        });
+        return _items;
+    }),
+    selectedUserItems: [] as SelectedUserItem[],
+    existingMemberList: [] as string[],
+    searchText: '',
+});
+
+/* Util */
+const fetchWorkspaceUsers = async () => {
+    try {
+        const response = await SpaceConnector.clientV2.identity.workspaceUser.list<WorkspaceUserListParameters, ListResponse<WorkspaceUserModel>>();
+        state.workSpaceUserList = response.results ?? [];
+    } catch (e) {
+        state.workSpaceUserList = [];
+        ErrorHandler.handleError(e);
+    }
+};
+const getRoleType = (userId: string): string|undefined => {
+    if (isEmpty(storeState.users)) return undefined;
+    return storeState.users[userId].data.roleInfo?.role_type;
+};
+
+/* Api */
+const addMember = async () => {
+    try {
+        const params: ProjectAddUsersParameters = {
+            project_group_id: props.projectGroupId,
+            users: state.selectedUserItems.map((d) => d.name),
+        };
+        await SpaceConnector.clientV2.identity.projectGroup.addUsers(params);
```

Comment:
nitpick: schema type is missing!

Reply:
I couldn't add the type yet because there's no `addUsers` verb in the project group API😅